samtranslator/utils/utils.py

```diff
@@ -0,0 +1,23 @@
+import copy
+from typing import cast, Any, List
+
+
+def as_array(x: Any) -> List[Any]:
+    """Convert value to list if it already isn't."""
+    return x if isinstance(x, list) else [x]
+
+
+def insert_unique(xs: Any, vs: Any) -> List[Any]:
+    """
+    Return copy of `xs` extended with values of `vs` that do not exist in `xs`.
+
+    Inputs are converted to lists if they already aren't.
+    """
+    xs = as_array(copy.deepcopy(xs))
+    vs = as_array(copy.deepcopy(vs))
+
+    for v in vs:
```

Comment:
Maybe I'm misreading the logic but can't you convert to sets and use the .union function

Reply:
Could, but we want it to be always predictable and deterministic (ensures we don't cause changes/redeployments and tests keep working regardless of Python version, and we can determine order without knowing implementation details).

For example:

```python
>>> list(set(['a', 'wat', 'lol']))
['lol', 'wat', 'a']
```

The order changes due to implementation details (the hashing); we don't want to rely on it staying as-is (in fact during the Python 2 transition to Python 3, there was a lot of work needed to keep code that relied on the underlying hashing algorithm backwards compatible).

Reply:
Ah I see, I thought .union had a guaranteed order. Thanks for the clarification